lib/filewatch/watch.rb

```diff
@@ -62,62 +45,66 @@ def initialize(opts={})
       end
       @watching = []
       @exclude = []
-      @files = Hash.new { |h, k| h[k] = WatchedFile.new(k, false, false) }
-      @unwatched = Hash.new
+      @files = Hash.new
       # we need to be threadsafe about the mutation
       # of the above 2 ivars because the public
-      # methods each, discover, watch and unwatch
+      # methods each, discover and watch
       # can be called from different threads.
       @lock = Mutex.new
       # we need to be threadsafe about the quit mutation
       @quit = false
       @quit_lock = Mutex.new
+      @max_active = ENV.fetch("FILEWATCH_MAX_OPEN_FILES", 4095).to_i
+      @max_warn_msg = "Reached maximum (#{@max_active}) open files"
+      @lastwarn_max_files = 0
     end # def initialize
 
     public
+
+    def max_open_files=(value)
+      val = value.to_i
+      return if value.nil? || val <= 0
+      @max_warn_msg = "Reached maximum (#{val}) open files"
+      @max_active = val
+    end
+
+    def ignore_older=(value)
+      #nil is allowed but 0 and negatives are made nil
+      if !value.nil?
+        val = value.to_f
+        val = val <= 0 ? nil : val
+      end
+      @ignore_older = val
+    end
+
+    def close_older=(value)
+      if !value.nil?
+        val = value.to_f
+        val = val <= 0 ? nil : val
+      end
+      @close_older = val
+    end
+
     def exclude(path)
       path.to_a.each { |p| @exclude << p }
     end
 
-    public
     def watch(path)
       synchronized do
         if !@watching.member?(path)
           @watching << path
           _discover_file(path) do |filepath, stat|
-            WatchedFile.new_initial(filepath, inode(filepath, stat))
+            WatchedFile.new_initial(
+                filepath, inode(filepath, stat), stat
+              ).init_vars(@delimiter, @ignore_older, @close_older)
```

Comment:
Feels like a builder pattern would clean this up:

WatchedFileBuilder.new.tap { |w| w.filepath = ...; w.delimiter = ... }.build

but we don't necessarily need to refactor that right now.